runners/spark/src/main/java/org/apache/beam/runners/spark/translation/streaming/UnboundedDataset.java

```diff
@@ -57,15 +60,20 @@ public UnboundedDataset(JavaDStream<WindowedValue<T>> dStream, List<Integer> str
     return streamSources;
   }
 
-  public void cache() {
-    dStream.cache();
+  @SuppressWarnings("unchecked")
+  public void cache(Coder<?> coder) {
+    // Caching can cause Serialization, we need to code to bytes
```

Comment:
Perhaps add link to the JIRA ticket explaining the problem? Also in `BoundedDataset`

Reply:
right